clang/lib/Sema/SemaChecking.cpp

```diff
@@ -5967,8 +5967,10 @@ void Sema::checkCall(NamedDecl *FDecl, const FunctionProtoType *Proto,
     CheckSYCLKernelCall(FD, Range, Args);
 
   // Diagnose variadic calls in SYCL.
-  if (FD && FD ->isVariadic() && getLangOpts().SYCLIsDevice &&
-      !isUnevaluatedContext() && !isKnownGoodSYCLDecl(FD))
+  if (FD && FD->isVariadic() &&
+      FD->getBuiltinID() != Builtin::BI__builtin_assume_aligned &&
```

Comment:
I wonder if it will be better to put check on `BI__builtin_assume_aligned` ID to `isKnownGoodSYCLDecl` function?
`isKnownGoodSYCLDecl` seems to be an intended function for such purposes.

Reply:
Sorry, just seeing your review comments.